### ISSUE TITLE
Loss and loops

### ISSUE THREAD

**Issue by WhisperWS** [mindflow-institue/DermoSegDiff#1]
How many times have you trained to get good results and how small should the loss be?

**Comment by afshinbigboy** [comment]
Regarding the value of the total loss, it depends on the determined coefficient values and parameters for different losses. However, to make sense of the training process, I have attached the change of `l2` loss per each epoch on PH2 and ISIC2018 datasets. (for both `BatchSize: 32` and `opt: Adam, lr=0.0001` )

<img width=400 src="https://github.com/xmindflow/DermoSegDiff/assets/6207884/72ed8200-c7d5-4a1e-85b8-77d1c02f95fc"/>

<img width=400 src="https://github.com/xmindflow/DermoSegDiff/assets/6207884/4b015521-3352-4a77-ad15-c12dabe86e1e"/>

_The graph with magenta (`tr`) and green (`val`) lines with about 360 epochs is for **ISIC2018**, and the graph with blue (`tr`) and red (`val`) with 3K epochs is for **PH2**. **HAM10000** is as same as this with fewer epochs._

----

To achieve good results with fewer parameters, you can also explore various configurations on your dataset. For instance, I have provided some example alternative configurations below that you can evaluate. You should fine-tune these configurations in conjunction with adjusting the training loss parameters. In my experiments with a `size` of 128, it is essential to have a minimum of 128 channels for the `x` in the primary layers.

| Configs (`cfg.model.params`) |     scenario 1     |     scenario 2     |  scenario 3  |   scenario 4     |    scenario ...  |
|-------------------------------|----------------|-----------------|-------------|---------------|--------------|
| `dim_x`                                    | 128                     | 128                    | 128              | 128                   | >= 128            |
| `dim_g`                                    | 64                      | 32                      | 64                | 32                     | >= 32             |
| `dim_x_mults`                         | [1,2,3,4,5,6]       | [1,2,3,4,5,6]       | [1,1,2,2,3,3]  | [2,2,2,2,2,2]    | ...                    |
| `dim_g_mults`                         | [1,2,4,8,16,32]   | [1,2,4,8,16,32]   | [1,2,3,4,5,6]  | [1,2,4,6,10,16] | ...                    |
